### PR TITLE
Add stylelint plugins to lint Design System typography, spacing and color rules

### LINKED ISSUE #125: Provide CSS linting of key rules
_As a user of the Design System, 
I want the key rules to be enforced with linting
So that I know that I'm using them correctly_

## Description

We currently use stylelint to lint our CSS. The rules are mainly used for consistent code and to minimise complexity (by whitelisting only a subset of CSS language). Only one stylelint rule is used to enforce a specific rule of the DS: `declaration-property-unit-whitelist` is used to enforce `rems` for type and spacing.

There's scope to add more rules for spacing, type and colour.

## Acceptance criteria

Add rules for:

- [ ] the value of `font-size` property (or font size within the `font` shorthand property) must use a slot
- [ ] the value of `line-height` property (or line height within the `font` shorthand property) must by a rem number divisible by quarter.
- [ ] the value of `margin`, `border` and `padding`, `grid-gap` properties (and their long hand versions) must by a rem number divisible by half.
- [ ] hex colours should only be brand ones
- [ ] `shade` and `tint` functions should be 10% steps

### PR DESCRIPTION
**Pull request by jeddy3**
Two commits, the first adds the plugins and the second updates the code to adhere to them.

Closes #125

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,14 +4,10 @@
...
   "scripts": {
-    "check-format":
-      "prettier '**/*.{css,json,js,md}' '!package.json' '!.github/*.md' --list-different --ignore-path .gitignore",
-    "format":
-      "prettier '**/*.{css,json,js,md}' '!package.json' '!.github/*.md' --write --ignore-path .gitignore",
-    "lint:css":
-      "stylelint '**/*.css' --max-warnings=0 --ignore-path .gitignore",
-    "lint:md":
-      "remark . --quiet --frail --rc-path .remarklintrc.json --ignore-path .gitignore",
+    "check-format": "prettier '**/*.{css,json,js,md}' '!package.json' '!.github/*.md' --list-different --ignore-path .gitignore",
```
Comment (joelcarr) [inline package.json]: Every time I edit `package.json`, prettier reformats it to drop all the values to the next line. How can we get around these changes if they are to be inline?

File: tools/stylelint/stylelint-color-hex-whitelist/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable import/unambiguous */
```
Comment (joelcarr) [inline tools/stylelint/stylelint-color-hex-whitelist/__tests__/index.js]: We should create a nested `.eslint` file to ensure we don't have to remember to write this comment on every new file.

Just like [here](https://github.com/barnardos/consent-form-builder/blob/apply-design-system/config/webpack/.eslintrc).

File: .stylelintrc.json [hunk 3]
```diff
@@ -1,7 +1,38 @@
 {
   "extends": "stylelint-config-recommended",
-  "plugins": ["stylelint-order", "stylelint-selector-bem-pattern"],
+  "plugins": [
+    "./tools/stylelint/index.js",
```
Comment (joelcarr) [inline .stylelintrc.json]: I have changed my mind from our chat about the naming of the `tools/` directory. We should place the `stylelint/` directory at the root of the repo. If we then come to a time where we need to create custom front end processing tools, then we can think about putting the code in a `tool/` like directory.

File: package.json (hunk 1 above)
Comment (jeddy3) [reply package.json]: This reverts it to the correct format (i.e. the one autoformatted by yarn). Going forward we need to find a way to configure the VSCode prettier extension to ignore this file.

File: tools/stylelint/stylelint-color-hex-whitelist/__tests__/index.js (hunk 2 above)
Comment (jeddy3) [reply tools/stylelint/stylelint-color-hex-whitelist/__tests__/index.js]: 👍

File: .stylelintrc.json (hunk 3 above)
Comment (jeddy3) [reply .stylelintrc.json]: 👍

**Comment by jeddy3** [comment]
Deploy preview - https://5ae840b0b13fb11505435ffb--barnardos-design-system.netlify.com/

For something reason it wasn't reported to GitHub.

**Comment by tosbourn** [comment]
> Looks good LGTM to me!

This is the _only_ acceptable use of the `LGTM` acronym.

**Pull request merged**